userspace/libsinsp/plugin_table_api.cpp

```diff
@@ -145,7 +145,7 @@ static inline owned_table_input_t copy_and_check_table_input(const sinsp_plugin*
 	res->reader_ext = new ss_plugin_table_reader_vtable_ext();
 	res->writer_ext = new ss_plugin_table_writer_vtable_ext();
 	res->fields_ext = new ss_plugin_table_fields_vtable_ext();
-	if (p->required_api_version().m_version_minor < 1)
+	if (p->required_api_version().minor() < 1)
```

Comment:
the change is OK! but I have some doubts about the logic behind it, what is the meaning of checking only against a `minor`? `3.0.1` will be valid  as `2.0.1` and `1.0.1`... am i missing something here? @jasondellaluce

Reply:
This is not related at all to this patch.

When bumping the plugin API from `3.0.0` to `3.1.0` there were some "design flaws" in the vtable that prevented adding new API symbols. However, changing that would have been a breaking change that would have made the plugin API jump to `4.0.0` just few months after releasing a Falco with `3.0.0`. Since we don't wanna keep doing this with users this frequently, the approach has been to add an "extended" vtable that now supports being extended (see how creative I've been with the naming :D). So this trick makes sure that for the currently-supported major, we respect backward compatibility with minor version less than 1 (v3.0.0) by making sure the non-extended and extended vtables are both filled-up and consistent.